esp/platform/espcfg.cpp

```diff
@@ -325,6 +326,10 @@ CEspConfig::CEspConfig(IProperties* inputs, IPropertyTree* envpt, IPropertyTree*
         if (m_cfg->getProp("@daliServers", daliservers))
             initDali(daliservers.str()); //won't init if detached
 
+        const unsigned dafilesrvConnectTimeout = m_cfg->getPropInt("@dafilesrvConnectTimeout", 2000);
```

Comment:
How were the defaults decided on?  Will a change in default behavior catch the users off guard?  Looks like old behavior was the equivalent of 0's here?

Reply:
We discussed the default timeout period last Monday.  @jakesmith suggested that having shorter timeout period would mean be sensible (resolves issues such as https://track.hpccsystems.com/browse/HPCC-19002) and if it were to cause problems, the timeout period could be extended with the configuration option. 

I agree that it would cause problems if say a 30 second timeout is necessary and eventually returns a results.  But does any remote file access require this length of time?  May be 2 seconds timeout is too short?

@jakesmith ?

Reply:
My suggestion was that global default (not just in esp) should be much shorter than it is now.
( Currently the defaults are 100 seconds and 3 retries. )
And it should be globally configurable (e.g. via environment.conf), so that if need / if problematic, it can be raised.

I think I had suggested changing it to 10 seconds (3 retries may be still ok).

Once that is done, whilst having a separate configuration in Esp is okay, it becomes less relevant.

There is also the further improvement of implementing a better mechanism via a timed blacklist, so clients don't keep retrying the same endponit and repeatedly timing out, which cumulatively can be significant even if short. (There is an implementation like this in thorsoapcall, which should be looked at to see whether it can be reused).
@shamser - was a JIRA opened for the blacklist implementation ?

Reply:
@jakesmith I have just created the blackout jira: https://track.hpccsystems.com/browse/HPCC-21927. @afishbeck Do you think it should be set to '0' or is 10 seconds ok?

Reply:
>I have just created the blackout jira
@shamser - blacklist?

Reply:
> Do you think it should be set to '0' or is 10 seconds ok?

@shamser - are you asking re. this esp custom timeout, or the global timeout I was talking about above?
Has a JIRA been opened to change the global timeout?

If the global timeout is set as suggested (10 secs) - then I'd  default the esp timeout to unset , will default to global timeout - but they (ops) can override it if they want to.

Reply:
@jakesmith I'm talking about the ESP custom and default timeouts.  This one is designed to address a number of issues reported relating to ESP services (making remote file requests) taking too long to timeout.  As far as other aspects related to timeouts, I feel that i may not be immediately pressing as no has complained (yet) and we can merge this one until we have agreed the more comprehensive approach for timings.   By the way, there was a bug in the previous commit, which I have now fixed and pushed. (I've also set the default timeout period to 10 seconds).